test/pinning_test.jl

```diff
@@ -132,3 +132,16 @@ end
     pinthreads(reverse(cpuids); force = false)
     @test getcpuids() == cpuids
 end
+
+@testset "Thread Pinning (random)" begin
+    pinthreads(getcpuids())
```

Comment:
Should be `cpuids_all()` rather than `getcpuids()` here (at least to semantically get what you had before). The former gives you the (static) numbering of cpuids on the system whereas the latter gives you the (dynamic) cpu ids where Julia threads are currently running.

(BTW, `pinthreads(getcpuids())` is essentially `pinthreads(:current)`.)

Reply:
Maybe it's worth reintroducing `pinthreads(:firstn)` to make this simpler but let's to this in a separate PR.  (#30)

Reply:
Ah I didn't realize that. My thinking was that it was simpler than doing e.g. `cpuids_all[1:Threads.nthreads()]`; but come to think of it, the test doesn't really care about the initial thread pinning so I just removed that line in adc7980 :shrug: